os/mm/mm_heap/mm_malloc.c

```diff
@@ -108,6 +112,18 @@ FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size, mmaddress_t caller_
 FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size)
 #endif
 {
+#ifdef CONFIG_BOOTMEM
+	if (g_bootmem) {
+		void *ret = g_bootmem;
```

Comment:
How are the boundary conditions handled
1) At any instant if the size requested is more than available g_heapsize.

Reply:
I second Sharis comment. You may consider to provide this boundary validation